app/javascript/gobierto_data/webapp/components/commons/DownloadButton.vue

```diff
@@ -52,32 +73,68 @@ export default {
     arrayFormats: {
       type: Object,
       required: true
-    }
+    },
+    queryStored: {
+      type: String,
+      default: ""
+    },
   },
   data() {
     return {
-      labelDownloadData: "",
+      labelDownloadData: I18n.t("gobierto_data.projects.downloadData") || "",
       isHidden: true,
-      sqlfileCSV: '',
-      sqlfileXLSX: '',
-      sqlfileJSON: '',
-      titleDataset: ''
+      arrayFormatsQuery: {},
+      titleFile: ''
+    }
+  },
+  watch: {
+    queryStored(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.createObjectFormatsQuery(newValue)
+      }
     }
   },
   created() {
-    this.labelDownloadData = I18n.t("gobierto_data.projects.downloadData")
+    this.createObjectFormatsQuery(this.queryStored)
   },
   methods: {
     closeMenu() {
       this.isHidden = true
     },
-    updateCode(sqlQuery) {
-      // TODO: this method do nothing
-      const code = sqlQuery
-      const endPointSQL = `${baseUrl}/data.csv?sql=`
-      this.sqlfileCSV = `${endPointSQL}${code}&csv_separator=semicolon`
-      this.sqlfileXLSX = `${endPointSQL}${code}`
-      this.sqlfileJSON = `${endPointSQL}${code}`
+    createObjectFormatsQuery(sql) {
+      //Convert all linebreaks from any SO(Windows: \r\n Linux: \n Older Macs: \r) to spaces.
+      const formatSQL = sql.replace(/[\r\n]+/gm, " ");
+      let datetime = new Date();
+      const date = `${datetime.getDate()}_${(datetime.getMonth() + 1)}_${datetime.getFullYear()}`;
+      const endPointCSV = `${baseUrl}/data.csv?sql=${formatSQL}&csv_separator=semicolon`
+      const endPointJSON = `${baseUrl}/data.json?sql=${formatSQL}`
+      const endPointXLSX = `${baseUrl}/data.xlsx?sql=${formatSQL}`
+
+      const {
+        params: {
+          id: titleFile
+        }
+      } = this.$route
+
+      this.titleFile = titleFile
+
+      this.arrayFormatsQuery = [
```

Comment:
Let's make the list of formats dependant on the values that the API returns. Formats are one of the attributes returned in the meta of a dataset:

![Screenshot 2020-05-07 at 17 32 58](https://user-images.githubusercontent.com/17616/81314078-f9da3280-9088-11ea-9bab-543ac4e224bf.png)

Reply:
We already talked about this in the past, I think...

Reply:
Done @ferblape